civis/tests/test_io.py

```diff
@@ -98,9 +99,11 @@ def setup_class(cls, *mocks):
 
     @mock.patch(api_import_str, return_value=civis_api_spec)
     def test_zip_member_to_civis(self, *mocks):
-        with tempfile.NamedTemporaryFile() as tmp:
-            with zipfile.ZipFile(tmp, 'w', zipfile.ZIP_DEFLATED) as zip_file:
-                zip_file.writestr(tmp.name, 'a,b,c\n1,2,3')
+        with TemporaryDirectory() as temp_dir:
+            fname = os.path.join(temp_dir, str(uuid.uuid4()))
+            with zipfile.ZipFile(fname, 'w', zipfile.ZIP_DEFLATED) as zip_file:
+                archive_name = str(uuid.uuid4())
+                zip_file.writestr(archive_name, 'a,b,c\n1,2,3')
```

Comment:
`archive_name` should be its own distinct filename rather than being the same as the zipf file's filename. Also, using the zip file's filename failed this test because apparently the `ZipFile` object normalizes backslashes into forward slashes in filenames internally, which caused test failures like [this](https://ci.appveyor.com/project/eabraun/civis-python/build/1.0.8/job/198d2m1dbp38fpy5).

Reply:
So this means that you can't use `archive_name = os.path.join(str(temp_dir, uuid.uiid4()))`?  If you run the tests locally, won't this litter your folder?

Reply:
I've just run the test suite locally to verify that no random temp files are created. `zip_file.writestr(archive_name, ...)` writes a new member file named `archive_name` to `zip_file` which is the already instantiated temp zip file; it doesn't mean creating a file called `archive_name` at the current directory.